src/lib.rs

```diff
@@ -41,63 +67,41 @@ pub fn derive(input: TokenStream) -> TokenStream {
         })
         .collect();
 
-    if let Some(idents) = get_from_ident(ast.attrs.iter()) {
-        if idents.len() < 1 {
-            panic!("No from type provided");
-        }
-        from_idents = idents;
-    /*
-    match idents.into_iter().nth(0) {
-        Some(first) => from_ident = first,
-        _ => panic!("No From attribute provided"),
-    }
-    */
+    let from_types;
+    if let Some(f) = get_from_types(ast.attrs.iter()) {
+        from_types = f;
     } else {
-        panic!("problem with from attribute");
+        panic!("error parsing the from types");
```

Comment:
write TODO: when the function get_from_types will return Result, use that for error message.